apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c

```diff
@@ -0,0 +1,145 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <pthread.h>
+#include <sys/time.h>
+#include <errno.h>
+#include <apps/shell/tash.h>
+#include <wifi_manager/wifi_manager.h>
+#include "../../../tc_common.h"
+
+void ocftest_wifi_sta_connected(void);
+void ocftest_wifi_sta_disconnected(void);
+void ocftest_wifi_softap_sta_join(void);
+void ocftest_wifi_softap_sta_leave(void);
+
+static wifi_manager_cb_s wifi_callbacks = {
+	NULL,						//ocftest_wifi_sta_connected,
+	NULL,						//ocftest_wifi_sta_disconnected,
+	NULL,						//ocftest_wifi_softap_sta_join,
+	NULL,						//ocftest_wifi_softap_sta_leave,
+};
+
+void utc_wifi_manager_init_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_init(NULL);
+
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_FAIL);
+	TC_SUCCESS_RESULT();
+}
+
+void utc_wifi_manager_init_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_init(&wifi_callbacks);
+
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+void utc_wifi_manager_set_mode_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_set_mode(SOFT_AP_MODE, NULL);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_FAIL);
+
+	wifi_manager_softap_config_s ap_config = {"PKES", 1, "1111122222"};
+
+	ret = wifi_manager_set_mode(WIFI_NONE, &ap_config);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_INVALID_ARGS);
+	TC_SUCCESS_RESULT();
+}
+
+void utc_wifi_manager_set_mode_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	wifi_manager_softap_config_s ap_config = {"PKES", 1, "1111122222"};
+	ret = wifi_manager_set_mode(SOFT_AP_MODE, &ap_config);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
+
+	ret = wifi_manager_set_mode(STA_MODE, NULL);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
+
+	TC_SUCCESS_RESULT();
+}
+
+void utc_wifi_manager_deinit_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_deinit();
+
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_FAIL);
+	TC_SUCCESS_RESULT();
+}
+
+void utc_wifi_manager_deinit_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_deinit();
+
+	TC_ASSERT_EQ("wifi_manager_deinit", ret, WIFI_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+int wifi_manager_utc(int argc, FAR char *argv[])
+{
+	total_pass = 0;
+	total_fail = 0;
+
+	utc_wifi_manager_init_n();
+	utc_wifi_manager_init_p();
+
+	utc_wifi_manager_deinit_p();
+	utc_wifi_manager_deinit_n(); // Should be run after positive tc
+
+	utc_wifi_manager_set_mode_n();
+	utc_wifi_manager_set_mode_p();
```

Comment:
in wifi_manager_set_mode(), there is a memory allocation : wifi_utils_start_soft_ap.
I think before finishing wifi_manager_utc, we should free those utils

Reply:
Inside wifi_utils_start_soft_ap(), two malloced variables are released before the fuction returns.
Please double-check this.